### DISCUSSION TITLE
Upload Bom: Whats the Token in Response for?

### DISCUSSION THREAD

**Discussion by JoergBruenner** [DependencyTrack/dependency-track#2199] [Q&A] [answered]
Hi all,

the "upload BoM"-API Call returns a token. It looks like an UUID. What is it used for?

Thanks a lot
Jörg

**Comment by nscuro** [comment, marked as answer]
It can be used to poll whether DT is still processing the BOM, using the `/api/v1/bom/token/<token>` endpoint.

Because BOM processing and analysis are asynchronous processes, DT can't return results immediately after the BOM has been uploaded.

Once the above endpoint returns `false`, it's safe to query the findings and policy violations APIs for results. As long as it returns `true`, clients may periodically poll the endpoint again until it returns `false`. This is useful in CI, where analysis results are required in order to make decisions on whether or not to fail the build etc.
